src/isl/mod.rs

```diff
@@ -67,43 +107,43 @@ mod isl_tests {
         load_named_type(r#" // For a schema with named type
                 type:: { name: my_int, type: int }
             "#),
-        IslType::NamedIslType(NamedIslType::new("my_int".to_owned(), vec![IslConstraint::Type(IslTypeRef::CoreIslType(IonType::Integer))]))
+        IslType::named("my_int".to_owned(), vec![IslConstraint::type_constraint(IslTypeRef::core_isl(IonType::Integer))])
     ),
     case::type_constraint_with_self_reference_type(
         load_named_type(r#" // For a schema with self reference type
                 type:: { name: my_int, type: my_int }
             "#),
-        IslType::NamedIslType(NamedIslType::new("my_int".to_owned(), vec![IslConstraint::Type(IslTypeRef::NamedType("my_int".to_owned()))]))
+        IslType::named("my_int".to_owned(), vec![IslConstraint::type_constraint(IslTypeRef::named("my_int".to_owned()))])
     ),
     case::type_constraint_with_nested_self_reference_type(
         load_named_type(r#" // For a schema with nested self reference type
                 type:: { name: my_int, type: { type: my_int } }
             "#),
-        IslType::NamedIslType(NamedIslType::new("my_int".to_owned(), vec![IslConstraint::Type(IslTypeRef::AnonymousType(AnonymousIslType::new(vec![IslConstraint::Type(IslTypeRef::NamedType("my_int".to_owned()))])))]))
+        IslType::named("my_int".to_owned(), vec![IslConstraint::type_constraint(IslTypeRef::anonymous(vec![IslConstraint::type_constraint(IslTypeRef::named("my_int".to_owned()))]))])
     ),
     case::type_constraint_with_nested_type(
         load_named_type(r#" // For a schema with nested types
                 type:: { name: my_int, type: { type: int } }
             "#),
-        IslType::NamedIslType(NamedIslType::new("my_int".to_owned(), vec![IslConstraint::Type(IslTypeRef::AnonymousType(AnonymousIslType::new(vec![IslConstraint::Type(IslTypeRef::CoreIslType(IonType::Integer))])))]))
+        IslType::named("my_int".to_owned(), vec![IslConstraint::type_constraint(IslTypeRef::anonymous(vec![IslConstraint::type_constraint(IslTypeRef::core_isl(IonType::Integer))]))])
     ),
     case::type_constraint_with_nested_multiple_types(
         load_named_type(r#" // For a schema with nested multiple types
                 type:: { name: my_int, type: { type: int }, type: { type: my_int } }
             "#),
-        IslType::NamedIslType(NamedIslType::new("my_int".to_owned(), vec![IslConstraint::Type(IslTypeRef::AnonymousType(AnonymousIslType::new(vec![IslConstraint::Type(IslTypeRef::CoreIslType(IonType::Integer))]))), IslConstraint::Type(IslTypeRef::AnonymousType(AnonymousIslType::new(vec![IslConstraint::Type(IslTypeRef::NamedType("my_int".to_owned()))])))]))
+        IslType::named("my_int".to_owned(), vec![IslConstraint::type_constraint(IslTypeRef::anonymous(vec![IslConstraint::type_constraint(IslTypeRef::core_isl(IonType::Integer))])), IslConstraint::type_constraint(IslTypeRef::anonymous(vec![IslConstraint::Type(IslTypeRef::named("my_int".to_owned()))]))])
```

Comment:
Making these methods generic over `Into<String>` and something similar for a `Vec` would eliminate a lot of `.to_owned()`s and `vec!`s.